cpp/src/cores/trim.cu

```diff
@@ -0,0 +1,133 @@
+                 out_one_core_flags.begin());
+
+    update_edge_src_property(
+      handle, graph_view, out_one_core_flags.begin(), edge_src_out_one_cores);
+    update_edge_dst_property(
+      handle, graph_view, out_one_core_flags.begin(), edge_dst_out_one_cores);
+
+    rmm::device_uvector<size_t> subgraph_offsets_out(2, handle.get_stream());
+    std::vector<size_t> h_subgraph_offsets_out{{0, out_one_core_flags.size()}};
+    raft::update_device(subgraph_offsets_out.data(),
+                      h_subgraph_offsets_out.data(),
+                      h_subgraph_offsets_out.size(),
+                      handle.get_stream());
+    handle.sync_stream();
+
+    auto [src, dst, wgt, offsets] = extract_induced_subgraphs(
+    handle,
+    graph_view,
+    edge_weight_view,
+    raft::device_span<size_t const>{subgraph_offsets_out.data(), subgraph_offsets_out.size()},
+    raft::device_span<vertex_t const>{out_one_core_flags.data(), out_one_core_flags.size()},
+    do_expensive_check);
+
+    // in degree 1 core
+    core_number(
+      handle, graph_view, core_numbers.data(), k_core_degree_type_t::IN, size_t{1}, size_t{1});
+
+    edge_src_property_t<decltype(graph_view), uint8_t> edge_src_in_one_cores(handle,
+                                                                                 graph_view);
+    edge_dst_property_t<decltype(graph_view), uint8_t> edge_dst_in_one_cores(handle,
+                                                                                 graph_view);
+    auto in_one_core_first =
+      thrust::make_transform_iterator(core_numbers.begin(), is_one_or_greater_t<edge_t>{});
+    rmm::device_uvector<uint8_t> in_one_core_flags(core_numbers.size(), handle.get_stream());
+    thrust::copy(handle.get_thrust_policy(),
+                 in_one_core_first,
+                 in_one_core_first + core_numbers.size(),
+                 in_one_core_flags.begin());
+    update_edge_src_property(
+      handle, graph_view, in_one_core_flags.begin(), edge_src_in_one_cores);
+    update_edge_dst_property(
+      handle, graph_view, in_one_core_flags.begin(), edge_dst_in_one_cores);
+    rmm::device_uvector<size_t> subgraph_offsets_in(2, handle.get_stream());
+    std::vector<size_t> h_subgraph_offsets_in{{0, in_one_core_flags.size()}};
+    raft::update_device(subgraph_offsets_in.data(),
+                      h_subgraph_offsets_in.data(),
+                      h_subgraph_offsets_in.size(),
+                      handle.get_stream());
+    handle.sync_stream();
+
+   
+    auto [src_, dst_, wgt_, offsets_] = extract_induced_subgraphs(
+    handle,
+    graph_view,
+    edge_weight_view,
+    raft::device_span<size_t const>{subgraph_offsets_in.data(), subgraph_offsets_in.size()},
+    raft::device_span<vertex_t const>{in_one_core_flags.data(), in_one_core_flags.size()},
+    do_expensive_check);
+    
+    return  std::make_tuple(std::move(src_), std::move(dst_), std::move(wgt_));
```

Comment:
And if you have worked on the new out-degree one core edges, vertex IDs are based on the (renumbered-)new graph. You need to unrenumber to get the original vertex IDs.

You can call `unrenumber_local_int_edges` (https://github.com/rapidsai/cugraph/blob/branch-23.06/cpp/include/cugraph/graph_functions.hpp#L270) for this.

Reply:
Let's use `unrenumber_int_vertices` (two times for source and destination vertex IDs) instead of `unrenumber_local_int_edges` and add a FIXME statement saying using `unrenumber_local_int_edges` can be more efficient.

Calling `unrenumbering_local_in_edges` requires some low-level knowledge about cuGraph internals (https://github.com/rapidsai/cugraph/blob/branch-23.06/cpp/src/structure/decompress_to_edgelist_impl.cuh#L163) and I think exposing this much low level details to the user code is not appropriate. We need to update  `unrenumbering_local_in_edges` to make this easier to use.